crates/node-db/src/lib.rs

```diff
@@ -553,7 +600,7 @@ pub fn list_unchecked_blocks(
         };
 
         // Add the solution.
-        let solution: Solution = decode(&solution_blob)?;
+        let solution = get_solution(tx, &solution_addr)?;
```

Comment:
and here. We should actually make an issue for this as it could be pretty slow for listing blocks.